### ISSUE TITLE
Product variants missing in cart tag

### ISSUE THREAD

**Issue by JanJanssensDigiti** [duncanmcclean/simple-commerce#328]
Here I am again :grimacing:. Found another bug:

## How to reproduce

When you have product variants and add them to your cart you can't acces the variant value therefore this value can't be stored in your order when you complete your order. Did some digging already and found that the Statamic value() function removes the variant value when calling the cart (items).

Tags/CartTags.php:
`$this->getOrMakeSessionCart()->entry()->toAugmentedArray()['items']->value()`

In short: The variant value is present in the Cart object but just get changed when you call it

I would have tried to solve it for you but I'm to new with statamic and laravel package development to be usefull :sweat_smile:

## Environment

Statamic 3.0.36 Pro
Laravel 8.20.1
PHP 7.4.13
doublethreedigital/simple-commerce 2.1.24

**Comment by duncanmcclean** [comment]
Thanks for reporting! Don't worry about 😅  

I actually have a plan of how to fix this, which would mean building a fieldtype just for selecting a variant. I was thinking of releasing it as part of v2.2 but I don't have a timescale for that just yet.

I could try and build a bare bones variant fieldtype for the time being if that would be helpful?

The fieldtypes are what control the 'augmentation', which you can learn more about over [on the Statamic documentation](https://statamic.dev/extending/augmentation#content).

**Comment by JanJanssensDigiti** [comment]
Interesting gonna look into that!

If it doesn't take alot of your time it could be helpfull otherwise we will just restructure the product data to not use variants for the time being.

**Comment by duncanmcclean** [comment]
> Interesting gonna look into that!
> 
> If it doesn't take alot of your time it could be helpfull otherwise we will just restructure the product data to not use variants for the time being.

I'll build one for you after work 😄

**Comment by duncanmcclean** [comment]
Turns out a fieldtype doesn't work which is sad 😓 

However, I've built a little tag so you can get a particular variant from a product. Just pop this below code into an `App\Tags\VariantInfo.php` file.

```php
<?php

namespace App\Tags;

use Statamic\Facades\Entry;
use Statamic\Tags\Tags;

class VariantInfo extends Tags
{
    protected static $handle = 'variant-info';

    // {{ variant-info product="product-id" variant="variant-key" }} {{ key }} - {{ variant }} {{ /variant-info }}
    public function index()
    {
        if (! $this->params->has('product') && ! $this->params->has('variant')) {
            throw new \Exception("You must pass in both a `product` and a `variant` to the variant-info tag for it to function");
        }

        $product = Entry::find($this->params->get('product'));
        $variantKey = $this->params->get('variant');

        if (! $product) {
            throw new \Exception("No product entry could be found");
        }

        if (! $product->has('product_variants')) {
            throw new \Exception("$product->id does not have any product variants");
        }

        $augmentedOptions = $product->augmentedValue('product_variants')->value()['options'];

        return collect($augmentedOptions)
            ->where('key', $variantKey)
            ->first();
    }
}
```

And.. you should be able to use it like this. You'll need to pass in both the product ID and the key of the specific variant you want to get:

```
{{ variant-info product="6c38ff1f-dbe6-45e9-81a9-b17d34294db3" variant="Red_Large" }} 
  {{ variant }} - {{ price }} 
{{ /variant-info }}
``` 

Hopefully that's helpful for you. Let me know if you have any questions.

**Comment by JanJanssensDigiti** [comment]
At the time of submitting this issue the cart object wouldn't give me the variant if one was selected. Suddenly this isn't a problem anymore... Might be a caching problem. So the initial problem is solved!

I'm using the tag you made in the cart to extract the right price if a variant is present. I also made a seperate tag to get a specific order so you can show the details of it on a order confirmation page where your tag serves the same purpose.

This tag above and an order tag might be a nice to have to come out-of-the-box with Simple Commerce. 😉
Thanks again for the quick response!

**Comment by duncanmcclean** [comment]
Awesome! I'm planning on doing it a bit differently in v2.2 but it should work better in there.

**Comment by duncanmcclean** [comment]
I've changed my mind. I've just tagged [a release](https://github.com/doublethreedigital/simple-commerce/releases/tag/v2.1.34) which adds a new 'product variant' fieldtype.

You should be able to swap it out with the variant fieldtype in your order blueprint. Bear in mind: the augmentation will only work for new orders or orders that are re-saved in the CP.
